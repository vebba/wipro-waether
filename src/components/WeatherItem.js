import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import Time from "./Time";
import Temperature from "./Temperature";
import Description from "./Description";
import WeatherItemHeader from "./WeatherItemHeader";

const WeatherItem = ({ predictions }) => {
  return (
    <div className="App-weatherItem">
      <WeatherItemHeader title={predictions[0].name || "Title"} />
      {predictions.map(item => (
        <div key={item.dt} className="prediction">
          <Time time={moment(item.dt_txt).format("HH:mm")} />
          <Temperature temperature={Math.floor(item.main["temp"])} />
          <Description description={item.weather[0]["main"]} />
        </div>
      ))}
    </div>
  );
};

WeatherItem.propTypes = {
  predictions: PropTypes.array
};
WeatherItem.defaultProps = {
  predictions: [
    {
      name: "DayName",
      dt: "3",
      main: {
        temp: "10"
      },
      weather: [
        {
          main: "Weather"
        }
      ]
    }
  ]
};
export default WeatherItem;
