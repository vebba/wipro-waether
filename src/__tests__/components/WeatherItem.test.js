import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import WeatherItem from "../../components/WeatherItem";

Enzyme.configure({ adapter: new Adapter() });

it("renders without data", () => {
  const wrapper = shallow(<WeatherItem />);
  expect(wrapper).toMatchSnapshot();
});

it("renders WeatherItem with data", () => {
  const predictions = [
    {
      name: "Day",
      dt: 1406106000444,
      main: {
        temp: 298.77,
        temp_min: 298.77,
        temp_max: 298.774,
        pressure: 1005.93,
        sea_level: 1018.18,
        grnd_level: 1005.93,
        humidity: 87,
        temp_kf: 0.26
      },
      weather: [
        {
          id: 804,
          main: "Clouds",
          description: "overcast clouds",
          icon: "04d"
        }
      ],
      clouds: { all: 88 },
      wind: { speed: 5.71, deg: 229.501 },
      sys: { pod: "d" },
      dt_txt: "2014-07-23 09:00:00"
    }
  ];
  const wrapper = shallow(<WeatherItem predictions={predictions} />);
  expect(wrapper).toMatchSnapshot();
});
