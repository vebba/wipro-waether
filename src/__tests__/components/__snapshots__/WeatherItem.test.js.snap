// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WeatherItem with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherItem
    predictions={
      Array [
        Object {
          "clouds": Object {
            "all": 88,
          },
          "dt": 1406106000444,
          "dt_txt": "2014-07-23 09:00:00",
          "main": Object {
            "grnd_level": 1005.93,
            "humidity": 87,
            "pressure": 1005.93,
            "sea_level": 1018.18,
            "temp": 298.77,
            "temp_kf": 0.26,
            "temp_max": 298.774,
            "temp_min": 298.77,
          },
          "name": "Day",
          "sys": Object {
            "pod": "d",
          },
          "weather": Array [
            Object {
              "description": "overcast clouds",
              "icon": "04d",
              "id": 804,
              "main": "Clouds",
            },
          ],
          "wind": Object {
            "deg": 229.501,
            "speed": 5.71,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WeatherItemHeaeder
          title="Day"
        />,
        Array [
          <div
            className="prediction"
          >
            <Time
              time="09:00"
            />
            <Temperature
              temperature={298}
            />
            <Description
              description="Clouds"
            />
          </div>,
        ],
      ],
      "className": "App-weatherItem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Day",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1406106000444",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Time
              time="09:00"
            />,
            <Temperature
              temperature={298}
            />,
            <Description
              description="Clouds"
            />,
          ],
          "className": "prediction",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "time": "09:00",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "temperature": 298,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "Clouds",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WeatherItemHeaeder
            title="Day"
          />,
          Array [
            <div
              className="prediction"
            >
              <Time
                time="09:00"
              />
              <Temperature
                temperature={298}
              />
              <Description
                description="Clouds"
              />
            </div>,
          ],
        ],
        "className": "App-weatherItem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Day",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1406106000444",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Time
                time="09:00"
              />,
              <Temperature
                temperature={298}
              />,
              <Description
                description="Clouds"
              />,
            ],
            "className": "prediction",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "time": "09:00",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "temperature": 298,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "Clouds",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherItem
    predictions={
      Array [
        Object {
          "dt": "3",
          "main": Object {
            "temp": "10",
          },
          "name": "DayName",
          "weather": Array [
            Object {
              "main": "Weather",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WeatherItemHeaeder
          title="DayName"
        />,
        Array [
          <div
            className="prediction"
          >
            <Time
              time="18:24"
            />
            <Temperature
              temperature={10}
            />
            <Description
              description="Weather"
            />
          </div>,
        ],
      ],
      "className": "App-weatherItem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "DayName",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Time
              time="18:24"
            />,
            <Temperature
              temperature={10}
            />,
            <Description
              description="Weather"
            />,
          ],
          "className": "prediction",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "time": "18:24",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "temperature": 10,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "Weather",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WeatherItemHeaeder
            title="DayName"
          />,
          Array [
            <div
              className="prediction"
            >
              <Time
                time="18:24"
              />
              <Temperature
                temperature={10}
              />
              <Description
                description="Weather"
              />
            </div>,
          ],
        ],
        "className": "App-weatherItem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "DayName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Time
                time="18:24"
              />,
              <Temperature
                temperature={10}
              />,
              <Description
                description="Weather"
              />,
            ],
            "className": "prediction",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "time": "18:24",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "temperature": 10,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "Weather",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
